components/feature/session/src/main/java/mozilla/components/feature/session/SessionUseCases.kt

```diff
@@ -174,6 +174,20 @@ class SessionUseCases(
         }
     }
 
+    /**
+     * Tries to recover from a crash by restoring the last know state.
+     */
+    class CrashRecoveryUseCase internal constructor(
+        private val sessionManager: SessionManager
+    ) {
+        fun invoke(session: Session) {
+            sessionManager.getOrCreateEngineSession(session)
+                .recoverFromCrash()
+
+            session.crashed = false
```

Comment:
What happens when we are unable to recover? Are we communicating a wrong message?

Reply:
I'll add the `Boolean` above.

Here in this case we tried to recover and that's all we could do. This doesn't mean the Session was fully restored - but we handled the crash. And then Session is operational and you can continue to load sites.

Reply:
👍